javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentStarterImpl.java

```diff
@@ -180,4 +206,117 @@ public void visitCode() {
       return hookInserted ? cw.toByteArray() : null;
     }
   }
+
+  private static class InetAddressClassFileTransformer implements ClassFileTransformer {
```

Comment:
Usually it is a good idea to keep the asm transforms as simple as possible because they are not easy to follow. I would change it so that it only instruments the `resolver` method. In that method search for the `isBooted` call (you already have this) and after it insert a call to a static method that returns boolean. So something like
```java
super.visitMethodInsn(opcode, ownerClassName, methodName, descriptor, isInterface);
if ("jdk/internal/misc/VM".equals(ownerClassName) && "isBooted".equals(methodName)) {
  super.visitMethodInsn(Opcodes.INVOKESTATIC,
    Type.getInternalName(AgentInitializer.class),
    "isAgentStarted",
    "(Z)Z",
    false);
}
```
`isAgentStarted` could be
```java
public static boolean isAgentStarted(boolean input) {
  return input && started;
}
```
This way you only need to add one instruction and don't need to recompute stack frames or maxs.

Reply:
done